astropy/io/fits/hdu/image.py

```diff
@@ -135,9 +147,31 @@ def __init__(self, data=None, header=None, do_not_scale_image_data=False,
                 self._data_needs_rescale = True
             return
         else:
+            # Setting data will set set _bitpix, _bzero, and _bscale to the
+            # appropriate BITPIX for the data, and always sets _bzero=0 and
+            # _bscale=1.
             self.data = data
             self.update_header()
```

Comment:
It looks to me like `update_header()` already gets called when setting `self.data`.  See https://github.com/mwcraig/astropy/blob/0627686c4c422c7f183e552d70d320a0f1540a5f/astropy/io/fits/hdu/image.py#L288

So this can probably be removed (if somehow things break unless it's called twice then that would be something to look into as well).

Basically most of this business of setting `self._bitpix` shouldn't be happening here in `__init__` at all, and should just happen when setting `self.data =...` (since that can happen outside of `__init__` as well).